### PR TITLE
Add to cart backend

### PR DESCRIPTION
**Pull request by mil-amirian**
Screen Shot 2020-09-29 at 11 26 13 AM
Screen Shot 2020-09-29 at 11 26 33 AM

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -61,6 +61,128 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+  } else {
+    const sql = `
+    select *
+      from "carts"
+  `;
+    db.query(sql)
+      .then(result => {
+        const cart = result.rows;
+        res.json(cart);
+      });
```
Comment (thebearingedge) [inline server/index.js]: The client should receive an empty array, no need to query the database

File: server/index.js [hunk 2]
```diff
@@ -61,6 +61,128 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+  }
+  const sql = `
+      select *
+        from "products"
+        where "productId" = $1
+    `;
+  db.query(sql, values)
+    .then(result1 => {
+      const price = result1.rows[0].price;
+      if (result1.rows.length < 1) {
```
Comment (thebearingedge) [inline server/index.js]: this `if` should come before trying to access `.price` on something that might not be there.

File: server/index.js [hunk 3]
```diff
@@ -61,6 +61,128 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+        return res.status(400).json({
+          error: 'Cart is empty'
+        });
```
Comment (thebearingedge) [inline server/index.js]: Here we need to `throw` a `new ClientError` as described in the instructions.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw

File: server/index.js [hunk 4]
```diff
@@ -61,6 +61,128 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+      req.session.cartId = cartId;
+      const updateCartItems =
+        `
+          insert into "cartItems" ("cartId", "productId", "price")
+          values ($1, $2, $3)
+          returning "cartItemId"
+          `;
+      const params = [cartId, bodyRequest, price];
+
+      db.query(updateCartItems, params)
```
Comment (thebearingedge) [inline server/index.js]: Be sure to `return` here in case the query fails. That way your final `.catch` can handle the error

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (mil-amirian) [reply server/index.js]: done

File: server/index.js (hunk 2 above)
Comment (mil-amirian) [reply server/index.js]: done

File: server/index.js (hunk 4 above)
Comment (mil-amirian) [reply server/index.js]: done

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**